modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -1194,4 +1234,35 @@ public void createApplication(Application.Name applicationName) {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<Application.Name, Set<String>> getTagListForApplications(Collection<Application.Name> applicationNames) {
+
+        LOGGER.debug("Retrieving Experiment Tags for applications {}", applicationNames);
+
+        try {
+            List<ListenableFuture<Result<ExperimentTagsByApplication>>> futures = new ArrayList<>();
+
+            for (Application.Name appName : applicationNames) {
```

Comment:
@abhishekjain88 Basically, in Cassandra, using IN query is ANTI design pattern and no matter number of rows it will be comparably better to use parallel point queries instead of single IN query... Plus, having IN query call available in Repository is risky if in future someone use that in the transaction-API calls (e.g. assignments, event-recording etc). 

About other suggestions, they may be better but would not be worth doing it... Considering that #1 this is admin-API call - low volume. #2 As long as it is a point query Cassandra should able to handle it in efficient way.

Regarding cache use, it is NOT good idea to use any kind of APP level caching for admin-API call..